### DISCUSSION TITLE
AIR 51.1.1.2 - Keyboard does not come out on many devices

### DISCUSSION THREAD

**Discussion by fix89** [airsdk/Adobe-Runtime-Support#3399] [General]
After updating our app with Air 51.1.1.2 we received numerous reports of users who did not see the keyboard when they touched an input text. We were forced to send a new update released with AIR 51.0.1.4 where the problem does not exist.

**Comment by ylazy** [comment]
Hi!
You should provide more details to determine where the problem is. What operating systems / devices are your users using? What framework are you using? What component is affected? Can you provide a sample code that can reproduce the problem?

**Reply by fix89** [reply]
It’s an Android only bug. I was unable to replicate the problem on my devices, but many of our app users reported the problem, many had a Samsung device.

**Reply by ajwfrost** [reply]
@fix89 we've not been able to reproduce this on Samsung devices here but we may not be checking the exact same thing. Can you confirm whether this is happening for TextField objects that are in 'input' mode, or for StageText objects? And do they say that the keyboard just doesn't appear, or might it be that it appears then immediately disappears? Plus: do you use the runtime in 'background' mode on Android?

If you can provide an APK file that we could use to reproduce this, it would be useful (along with whatever instructions are needed to get to the location where the keyboard should be triggered..)

thanks

**Comment by maxddd23** [comment]
I also encountered the same issue.

TextFiled requires multiple clicks to call up the keyboard.

The following videos are for AIR versions 51.0.1.3 and 51.1.1.3 respectively


51.0.1.3   AIR version 51.0.1.3 is working fine.

https://github.com/user-attachments/assets/fceea2a8-e216-4aac-8b53-4778edb99d73


51.1.1.3   AIR version 51.1.1.3 is calling keyboard difficult.

https://github.com/user-attachments/assets/03e18052-0877-4b29-ba73-40e6140ac8cf

Devices known to have this issue
1.Samsung Galaxy S21
2.Samsung Galaxy S23 Ultra
3.Samsung Galaxy Tab S7 FE

![S__28721159_0](https://github.com/user-attachments/assets/e60216f9-4314-49c0-abba-fa14bc2c44a0)
![S__28721157_0](https://github.com/user-attachments/assets/aa957914-5ce6-4250-924a-88e6229150e7)

Please check this issue, thank you.
If you need an apk file, I will make an additional one and provide it to you.

**Comment by fix89** [comment]
Any news on this? We can not use the latest AIR SDK

**Comment by fix89** [comment]
Latest AIR 5.1.1.4 fix this issue?

**Reply by ajwfrost** [reply]
I don't think we've yet managed to reproduce this one .. 
@maxddd23  you mentioned about creating an APK, is it possible to create something simple that will reproduce this?
We also had some queries earlier, so just to check:
* Can you confirm whether this is happening for TextField objects that are in 'input' mode, or for StageText objects?
* And is it that the keyboard just doesn't appear, or might it be that it appears then immediately disappears?
* Plus: do you use the runtime in 'background' mode on Android?

thanks

**Reply by maxddd23** [reply]
Hi!

Can you confirm whether this is happening for TextField objects that are in 'input' mode, or for StageText objects?

A:Yes, but i only use TextField(input).


And is it that the keyboard just doesn't appear, or might it be that it appears then immediately disappears?

A:The keyboard doesn't appear to appear.


Plus: do you use the runtime in 'background' mode on Android?

A:I didn't set it up specifically, I packaged it into the APK.


These input textfields have the following settings:

.mouseWheelEnabled = false;
.restrict = "^0-9 ";

Sample APK & Video (There are 4 input textfields in this sample)
[Sample_Video.zip](https://github.com/user-attachments/files/16973906/Sample_Video.zip)

thanks.

**Reply by ajwfrost** [reply]
Thanks for the sample. Reproduced it sporadically on a local S20, but not at all on an AWS S21... But just to check:
* you first saw this in AIR 51.1, and it was okay beforehand?
* can you see whether you think it's caused by a slight movement of the finger when clicking in the text field?

From what I can tell, we had made a change in 51.1 which may have this side effect, but programmatically the issue should only occur if you have a "touch down" immediately followed by a "touch move" in a different place. So if your finger is moving slightly, it's going to think you're trying to scroll the app and will forward the event on...

One option here would just be to adjust the sensitivity, so that it works unless you're significantly moving the touch point. I can perhaps get a version of the runtimeClasses.jar put together to test out this workaround... or else we can revert the earlier change and look at a different way to implement that functionality.

thanks

**Reply by fix89** [reply]
Hello! I can confirm that this issue has first occurred with the 51.1 version of AIR. It also makes sense what you say, because in almost all screens we use a scroller component where we are listening on the touch_down, which in turn activates the listening of the touch_move event

**Reply by maxddd23** [reply]
HI

I only listen to the MouseEvent.CLICK event.

I can confirm that this problem does not exist in 51.0.1.3.

So I wanted to confirm if this is a 51.1 specific issue.


I can't tell if it's caused by slight movement of my fingers.

Because sometimes the keyboard cannot be displayed even if you click hard.


This problem has caused considerable distress to some users.

My app does not use the scroller component, it is just a simple input TextField.

But it still happens.


If you can correct this issue or have better suggestions for optimization.

Please help me. Thank you.

**Reply by ajwfrost** [reply]
Thanks - just to confirm/clarify, when I was mentioning the touch down/move events, these are the direct events we are getting from the Android system and handling internally, rather than what we're passing on to the AS3 code (which would be independent of our changes here). So it's not related to which events you're listening to, but from the code changes that went in, it's related to what actually happens from the Android OS from the touch events.

It's currently actually a one-pixel sensitivity and probably the way some screens work, as a finger approaches the screen you'd get an additional touch point, but then if you press, the touch "point" given would be at the heaviest pressure point, which may then be different.

By the sounds of it though - I will revert the earlier change in its entirety, and we'll make sure we don't have a side-effect like this when we put back in the handling for pass-through scrolling (i.e. where you're trying to scroll the whole application and you happen to touch in a text field).

thanks

**Reply by maxddd23** [reply]
Thank you for your support.

**Reply by fix89** [reply]
Thank you! Do you have a date for the new AIR release? Or is it possibile to have a beta release to test?

**Reply by ajwfrost** [reply]
Hi - hopefully we will get our next release towards the end of next week, or early in the following week - but for now, the attached can be unzipped and placed over the 51.1.1.4 SDK, in /lib/android/lib folder.
[android_patch.zip](https://github.com/user-attachments/files/16995044/android_patch.zip)

thanks
